### ISSUE TITLE
Install with npm is failing due to not finding cdk which does exist

### ISSUE THREAD

**Issue by brennanMKE** [aws-amplify/amplify-cli#7654]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.4.1

### Amplify CLI Version

N/A

### What operating system are you using?

Mac

### Amplify Categories

storage

### Amplify Commands

status

### Describe the bug

The installer with npm keeps failing. I've updated node to the latest version and tried again. I have manually deleted the current amplify and cdk files. This command keeps failing.

```sh
> npm install -g @aws-amplify/cli
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /Users/stehlib/.npm-global/lib/node_modules/@aws-amplify/cli/node_modules/aws-cdk/bin/cdk
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/stehlib/.npm-global/lib/node_modules/@aws-amplify/cli/node_modules/aws-cdk/bin/cdk'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

### Expected behavior

The npm install shown in [the docs](https://docs.amplify.aws/cli/start/install) should complete successfully.

### Reproduction steps

1. npm install -g @aws-amplify/cli
2. Look for the error
3. 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by edwardfoyle** [comment]
@brennanMKE npm install is working for me. Some things you can try:
1. `sudo npm i -g @aws-amplify/cli` -- in general `sudo npm install ...` isn't recommended, but depending on how you have npm configured it's sometimes necessary
2. Check that the global installation location is owned by the current system user. The global installation location can be found using `npm root -g`
3. As a last resort, try reinstalling node & npm

**Comment by brennanMKE** [comment]
In my `~/.npmrc` I have just 1 line.

```
prefix=~/.npm-global
```

I never run `npm` with `sudo` since it should have access to install files in `~/.npm-global` which I have added to my `PATH`. I've updated the latest version of `node` already. I also ran `npm -g cache clean --force` and `npm cache clean --force` and tried the install again. It keeps failing with the same error.

Why is this `chmod` command failing with `cdk`? I am able to confirm the path exists after the install fails.
